### PR TITLE
feat: add support for resolving blob images

### LINKED ISSUE #1884: Allow to use Blob as the Image source
**Is your feature request related to a problem? Please describe.**
On web contexts it is more handy to be able to use a Blob as the data source.
Even on modern versions of Node and other runtimes like Deno, have implemented this Web API.

A common use case would be taking an image from a canvas:
```jsx
<Image
   src={() => {
     const canvas = document.createElement('canvas')
     renderQR(canvas, "Hello world")
     // Now what?
   }}
/>
```

A naive approach would be creating a Data url with `canvas.toDataURL()`. But it is known to have some [drawbacks](https://stackoverflow.com/questions/59020799/which-function-should-i-use-todataurl-or-toblob).

So. Why don't create a Blob and use URL.createObjectUrl?
```jsx
<Image
  src={async () => {
    const canvas = document.createElement("canvas");
    renderQR(canvas, "Hello world");
    const blob = await new Promise((resolve, reject) =>
      canvas.toBlob((blob) => {
        if (blob) {
          return resolve(blob);
        } else {
          return reject(new Error("couldn't create Blob"));
        }
      })
    );
    return URL.createObjectUrl(blob);
  }}
/>
```

Now we have a worse problem. **There is a memory leak!**. Since we are not calling URL.revokeObjectUrl to release the resources. We could workaround it with a `useRef` and a `useEffect`, but the code becomes very verbose and hard to follow.

**Describe the solution you'd like**
Allow to pass a Blob as the source for images, instead of only allowing Buffer, which is a Node-only API (we could polyfill it on other environments, but it just doesn't feels natural)

It would allow something like:
```jsx
<Image
  src={() => {
    const canvas = document.createElement("canvas");
    renderQR(canvas, "Hello world");
    return new Promise((resolve, reject) =>
      canvas.toBlob((blob) => {
        if (blob) {
          return resolve(blob);
        } else {
          return reject(new Error("couldn't create Blob"));
        }
      })
    );
  }}
/>
```

> As a sidenote I think this can be a starting point to reduce the dependencies on node specific APIs and replacing them by Web APIs, like Web Streams, the Fetch API, etc.

**Describe alternatives you've considered**
- Polyfilling Node buffer and convert the blob to that.
- Using URL.createObjectURL as pointed above.

**Additional context**

### PR DESCRIPTION
**Pull request by davbrito**
fixes #1884

### CODE REVIEW

**Comment by diegomura** [comment]
Thanks @davbrito ! Can you rebase? I'll take a look after but looking good

**Comment by davbrito** [comment]
> Thanks @davbrito ! Can you rebase? I'll take a look after but looking good

Done!

File: packages/image/tests/resolve.test.js [hunk 1]
```diff
@@ -162,4 +162,33 @@ describe('image resolveImage', () => {
...
+  test('Should render a blob image', async () => {
+    const blob = new Blob([localJPGImage], { type: 'image/jpeg' });
+    const image = await resolveImage(blob);
+
+    expect(image.data).toBeTruthy();
+    expect(image.width).toBeGreaterThan(0);
+    expect(image.height).toBeGreaterThan(0);
+  });
+
+  test('Should render a blob whithout type', async () => {
```
Comment (diegomura) [inline packages/image/tests/resolve.test.js:175]:
```suggestion
  test('Should render a blob without type', async () => {
```

**Review by diegomura — APPROVED** [review]
utACK

Thanks for this great contribution!
